### PR TITLE
replace promises with async/await in ./pages

### PR DESCRIPTION
**Pull request by GeoBrodas**
Hi @jenkoian !

This is a great project, and a little complex for me as well, so I didn't touch anything in the `./api` folder. I have replaced the promises I saw in the `hooks` folder. All you need to do now is check! Hope it works! ( fingers crossed )

### CODE REVIEW

File: src/pages/User/components/PullRequests/hooks/useFetchPullRequests.js [hunk 1]
```diff
@@ -8,10 +8,19 @@ const ACTIONS = {
...
 const API_URL = process.env.REACT_APP_API_URL;
 
-const fetchPullRequests = (username) =>
-  fetch(`${API_URL}/prs?username=${username}`, {
-    method: 'GET',
-  }).then((response) => response.json());
+// const fetchPullRequests = (username) =>
+//   fetch(`${API_URL}/prs?username=${username}`, {
+//     method: 'GET',
+//   }).then((response) => response.json());
```
Comment (jenkoian) [inline src/pages/User/components/PullRequests/hooks/useFetchPullRequests.js:14]: Just remove all this, don't worry about leaving it in commented out.

File: src/pages/User/components/PullRequests/hooks/useFetchPullRequests.js [hunk 2]
```diff
@@ -8,10 +8,19 @@ const ACTIONS = {
...
+
+// Code by Georgey
```
Comment (jenkoian) [inline src/pages/User/components/PullRequests/hooks/useFetchPullRequests.js:16]: Can you remove? You can add your name to https://github.com/jenkoian/hacktoberfest-checker/blob/master/public/humans.txt if you want additional credit.

File: src/pages/User/components/PullRequests/hooks/useFetchPullRequests.js [hunk 3]
```diff
@@ -34,19 +43,36 @@ export default function useFetchPullRequests(username) {
...
+    //       type: ACTIONS.PULL_REQUESTS_FETCHED,
+    //       payload: { pullRequests },
+    //     })
+    //   )
+    //   .catch((error) =>
+    //     dispatch({
+    //       type: ACTIONS.PULL_REQUESTS_FETCH_ERROR,
+    //       payload: { error },
+    //     })
+    //   );
```
Comment (jenkoian) [inline src/pages/User/components/PullRequests/hooks/useFetchPullRequests.js:58]: Just remove all this, as above.

File: src/pages/User/components/PullRequests/hooks/useFetchPullRequests.js [hunk 4]
```diff
@@ -34,19 +43,36 @@ export default function useFetchPullRequests(username) {
...
+
+    // code by Georgey
+    const response = await fetchPullRequests(username);
```
Comment (jenkoian) [inline src/pages/User/components/PullRequests/hooks/useFetchPullRequests.js:61]: This errors, it needs to be inside an async function. Can you take another look?

File: src/pages/User/components/PullRequests/hooks/useFetchPullRequests.js [hunk 5]
```diff
@@ -34,19 +43,36 @@ export default function useFetchPullRequests(username) {
...
+      type: ACTIONS.PULL_REQUESTS_FETCHED,
+      payload: {response},
+    })
+
+    const errors = dispatch({ 
+      type: ACTIONS.PULL_REQUESTS_FETCH_ERROR,
+      payload: {data}
+    })
+
+    if(errors) {
```
Comment (jenkoian) [inline src/pages/User/components/PullRequests/hooks/useFetchPullRequests.js:72]: Think we're missing a space in here, you should be able to run `npm run prettier-fix` to correct 👍

**Review by jenkoian — CHANGES_REQUESTED** [review]
Thanks for having a go at this, few comments 👍

**Comment by GeoBrodas** [comment]
Hi @jenkoian ! Did everything as requested! Found this on how to use `async` inside a `useEffect`, and that's how I implemented it.
https://stackoverflow.com/questions/56838392/how-to-call-an-async-function-inside-a-useeffect-in-react

**Comment by jenkoian** [comment]
Thanks @GeoBrodas I'm getting errors when I run it locally, are you able to take a look?

**Comment by GeoBrodas** [comment]
Can you check now?

**Comment by jenkoian** [comment]
@GeoBrodas same result I'm afraid

**Comment by GeoBrodas** [comment]
> @GeoBrodas same result I'm afraid

I'll try to work on it! Typical case, never encountered this.

**Pull request closed without merging**